CI: Performance build fails

[Build fails in the journal module ](https://github.com/camunda/zeebe/actions/runs/7612190618/job/20729263287)with:

```
Error:  Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.1.1:java (default) on project zeebe-journal: An exception occurred while executing the Java class. Message header not found in map for key: messageHeader -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.1.1:java (default) on project zeebe-journal: An exception occurred while executing the Java class. Message header not found in map for key: messageHeader
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:193)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:180)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
Caused by: org.apache.maven.plugin.MojoExecutionException: An exception occurred while executing the Java class. Message header not found in map for key: messageHeader
    at org.codehaus.mojo.exec.ExecJavaMojo.execute (ExecJavaMojo.java:354)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:193)
    at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:180)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:572)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
Caused by: java.lang.IllegalStateException: Message header not found in map for key: messageHeader
    at org.agrona.Verify.present (Verify.java:70)
    at uk.co.real_logic.sbe.xml.MessageSchema.<init> (MessageSchema.java:68)
    at uk.co.real_logic.sbe.xml.XmlSchemaParser.parse (XmlSchemaParser.java:149)
    at uk.co.real_logic.sbe.SbeTool.parseSchema (SbeTool.java:349)
    at uk.co.real_logic.sbe.SbeTool.main (SbeTool.java:250)
    at org.codehaus.mojo.exec.ExecJavaMojo.lambda$execute$0 (ExecJavaMojo.java:283)
    at java.lang.Thread.run (Thread.java:1583)
```

Cause is unclear, but fails the build on main.

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -2437,11 +2437,11 @@
             -->
             <execution>
               <id>generate-sbe</id>
-              <!-- not bound by default since it would execute in places where we don't want SBE generation -->
-              <phase>none</phase>
               <goals>
                 <goal>exec</goal>
               </goals>
+              <!-- not bound by default since it would execute in places where we don't want SBE generation -->
+              <phase>none</phase>
               <configuration combine.children="override">
                 <!--
                   we have to use the goal exec with the java executable as otherwise the process
@@ -2458,7 +2458,7 @@
                   <argument>-classpath</argument>
                   <!-- automatically creates the classpath using all project dependencies,
                      also adding the project build directory -->
-                  <classpath/>
+                  <classpath></classpath>
                   <argument>uk.co.real_logic.sbe.SbeTool</argument>
                 </arguments>
                 <workingDirectory>${project.build.directory}/generated-sources</workingDirectory>

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -2421,27 +2421,50 @@
           <groupId>org.codehaus.mojo</groupId>
           <artifactId>exec-maven-plugin</artifactId>
           <version>${plugin.version.exec}</version>
-          <configuration>
-            <!-- Define system properties in one place, see https://github.com/camunda-zeebe/zeebe/issues/377 -->
-            <systemProperties>
-              <systemProperty>
-                <key>sbe.output.dir</key>
-                <value>${project.build.directory}/generated-sources/sbe</value>
-              </systemProperty>
-              <systemProperty>
-                <key>sbe.java.generate.interfaces</key>
-                <value>true</value>
-              </systemProperty>
-              <systemProperty>
-                <key>sbe.decode.unknown.enum.values</key>
-                <value>true</value>
-              </systemProperty>
-              <systemProperty>
-                <key>sbe.xinclude.aware</key>
-                <value>true</value>
-              </systemProperty>
-            </systemProperties>
-          </configuration>
+          <dependencies>
+            <dependency>
+              <groupId>uk.co.real-logic</groupId>
+              <artifactId>sbe-tool</artifactId>
+              <version>${version.sbe}</version>
+            </dependency>
+          </dependencies>
+          <executions>
+            <!--
+              Use exec:exec to generate SBE instead of exec:java, as exec:java does not fork the
+              process, and the SBE tool is not thread safe, i.e. could not be used with a parallel
+              multi-module maven build.
+              See: https://github.com/camunda/camunda/issues/16029
+            -->
+            <execution>
+              <id>generate-sbe</id>
+              <!-- not bound by default since it would execute in places where we don't want SBE generation -->
+              <phase>none</phase>
+              <goals>
+                <goal>exec</goal>
+              </goals>
+              <configuration combine.children="override">
+                <!--
+                  we have to use the goal exec with the java executable as otherwise the process
+                  will not fork, and this can cause thread safety issues in a parallel multi-module
+                  build
+                -->
+                <executable>java</executable>
+                <includePluginDependencies>true</includePluginDependencies>
+                <arguments combine.children="append">
+                  <argument>-Dsbe.output.dir=${project.build.directory}/generated-sources/sbe</argument>
+                  <argument>-Dsbe.java.generate.interfaces=true</argument>
+                  <argument>-Dsbe.decode.unknown.enum.values=true</argument>
+                  <argument>-Dsbe.xinclude.aware=true</argument>
+                  <argument>-classpath</argument>
+                  <!-- automatically creates the classpath using all project dependencies,
+                     also adding the project build directory -->
+                  <classpath/>
+                  <argument>uk.co.real_logic.sbe.SbeTool</argument>
+                </arguments>
+                <workingDirectory>${project.build.directory}/generated-sources</workingDirectory>
+              </configuration>
+            </execution>
+          </executions>
         </plugin>
 
         <!-- Resources Plugin -->

--- a/zeebe/atomix/cluster/pom.xml
+++ b/zeebe/atomix/cluster/pom.xml
@@ -299,30 +299,16 @@
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>exec-maven-plugin</artifactId>
-        <configuration>
-          <includeProjectDependencies>false</includeProjectDependencies>
-          <includePluginDependencies>true</includePluginDependencies>
-          <mainClass>uk.co.real_logic.sbe.SbeTool</mainClass>
-          <arguments>
-            <argument>${project.build.resources[0].directory}/snapshot-schema.xml</argument>
-            <argument>${project.build.resources[0].directory}/raft-entry-schema.xml</argument>
-          </arguments>
-          <workingDirectory>${project.build.directory}/generated-sources</workingDirectory>
-          <!-- system properties defined in zeebe-parent -->
-        </configuration>
-        <dependencies>
-          <dependency>
-            <groupId>uk.co.real-logic</groupId>
-            <artifactId>sbe-tool</artifactId>
-            <version>${version.sbe}</version>
-          </dependency>
-        </dependencies>
         <executions>
           <execution>
-            <goals>
-              <goal>java</goal>
-            </goals>
+            <id>generate-sbe</id>
             <phase>generate-sources</phase>
+            <configuration>
+              <arguments>
+                <argument>${project.build.resources[0].directory}/snapshot-schema.xml</argument>
+                <argument>${project.build.resources[0].directory}/raft-entry-schema.xml</argument>
+              </arguments>
+            </configuration>
           </execution>
         </executions>
       </plugin>

--- a/zeebe/broker/pom.xml
+++ b/zeebe/broker/pom.xml
@@ -443,29 +443,15 @@
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>exec-maven-plugin</artifactId>
-        <configuration>
-          <includeProjectDependencies>false</includeProjectDependencies>
-          <includePluginDependencies>true</includePluginDependencies>
-          <mainClass>uk.co.real_logic.sbe.SbeTool</mainClass>
-          <arguments>
-            <argument>${project.build.resources[0].directory}/broker-protocol.xml</argument>
-          </arguments>
-          <workingDirectory>${project.build.directory}/generated-sources</workingDirectory>
-        </configuration>
-        <!-- system properties defined in zeebe-parent -->
-        <dependencies>
-          <dependency>
-            <groupId>uk.co.real-logic</groupId>
-            <artifactId>sbe-tool</artifactId>
-            <version>${version.sbe}</version>
-          </dependency>
-        </dependencies>
         <executions>
           <execution>
-            <goals>
-              <goal>java</goal>
-            </goals>
+            <id>generate-sbe</id>
             <phase>generate-sources</phase>
+            <configuration>
+              <arguments>
+                <argument>${project.build.resources[0].directory}/broker-protocol.xml</argument>
+              </arguments>
+            </configuration>
           </execution>
         </executions>
       </plugin>

--- a/zeebe/journal/pom.xml
+++ b/zeebe/journal/pom.xml
@@ -129,29 +129,15 @@
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>exec-maven-plugin</artifactId>
-        <configuration>
-          <includeProjectDependencies>false</includeProjectDependencies>
-          <includePluginDependencies>true</includePluginDependencies>
-          <mainClass>uk.co.real_logic.sbe.SbeTool</mainClass>
-          <arguments>
-            <argument>${project.build.resources[0].directory}/journal-schema.xml</argument>
-          </arguments>
-          <workingDirectory>${project.build.directory}/generated-sources</workingDirectory>
-          <!-- system properties defined in zeebe-parent -->
-        </configuration>
-        <dependencies>
-          <dependency>
-            <groupId>uk.co.real-logic</groupId>
-            <artifactId>sbe-tool</artifactId>
-            <version>${version.sbe}</version>
-          </dependency>
-        </dependencies>
         <executions>
           <execution>
-            <goals>
-              <goal>java</goal>
-            </goals>
+            <id>generate-sbe</id>
             <phase>generate-sources</phase>
+            <configuration>
+              <arguments>
+                <argument>${project.build.resources[0].directory}/journal-schema.xml</argument>
+              </arguments>
+            </configuration>
           </execution>
         </executions>
       </plugin>

--- a/zeebe/protocol/pom.xml
+++ b/zeebe/protocol/pom.xml
@@ -174,29 +174,16 @@
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>exec-maven-plugin</artifactId>
-        <configuration>
-          <includeProjectDependencies>false</includeProjectDependencies>
-          <includePluginDependencies>true</includePluginDependencies>
-          <mainClass>uk.co.real_logic.sbe.SbeTool</mainClass>
-          <arguments>
-            <argument>${project.build.outputDirectory}/protocol.xml</argument>
-            <argument>${project.build.outputDirectory}/cluster-management-protocol.xml</argument>
-          </arguments>
-          <workingDirectory>${project.build.directory}/generated-sources</workingDirectory>
-        </configuration>
-        <dependencies>
-          <dependency>
-            <groupId>uk.co.real-logic</groupId>
-            <artifactId>sbe-tool</artifactId>
-            <version>${version.sbe}</version>
-          </dependency>
-        </dependencies>
         <executions>
           <execution>
-            <goals>
-              <goal>java</goal>
-            </goals>
+            <id>generate-sbe</id>
             <phase>generate-sources</phase>
+            <configuration>
+              <arguments>
+                <argument>${project.build.outputDirectory}/protocol.xml</argument>
+                <argument>${project.build.outputDirectory}/cluster-management-protocol.xml</argument>
+              </arguments>
+            </configuration>
           </execution>
         </executions>
       </plugin>

--- a/zeebe/transport/pom.xml
+++ b/zeebe/transport/pom.xml
@@ -143,28 +143,15 @@
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>exec-maven-plugin</artifactId>
-        <configuration>
-          <includeProjectDependencies>false</includeProjectDependencies>
-          <includePluginDependencies>true</includePluginDependencies>
-          <mainClass>uk.co.real_logic.sbe.SbeTool</mainClass>
-          <arguments>
-            <argument>${project.build.outputDirectory}/stream-protocol.xml</argument>
-          </arguments>
-          <workingDirectory>${project.build.directory}/generated-sources</workingDirectory>
-        </configuration>
-        <dependencies>
-          <dependency>
-            <groupId>uk.co.real-logic</groupId>
-            <artifactId>sbe-tool</artifactId>
-            <version>${version.sbe}</version>
-          </dependency>
-        </dependencies>
         <executions>
           <execution>
-            <goals>
-              <goal>java</goal>
-            </goals>
+            <id>generate-sbe</id>
             <phase>generate-sources</phase>
+            <configuration>
+              <arguments>
+                <argument>${project.build.outputDirectory}/stream-protocol.xml</argument>
+              </arguments>
+            </configuration>
           </execution>
         </executions>
       </plugin>

